python/cli/calibrate_frd/calibrate_frd.py

```diff
@@ -0,0 +1,232 @@
+        hover_point = (x, y)
+
+
+def draw_rectigle(img, pixel, color):
+    x, y = pixel
+    CROSSHAIR_SIZE = 15
+    CENTER_GAP = 2
+    cv2.line(img,
+            (x, y - CROSSHAIR_SIZE),
+            (x, y - CENTER_GAP),
+            color, 1)
+    cv2.line(img,
+            (x, y + CENTER_GAP),
+            (x, y + CROSSHAIR_SIZE),
+            color, 1)
+    cv2.line(img,
+            (x - CROSSHAIR_SIZE, y),
+            (x - CENTER_GAP, y),
+            color, 1)
+    cv2.line(img,
+            (x + CENTER_GAP, y),
+            (x + CROSSHAIR_SIZE, y),
+            color, 1)
+
+
+class RayApp:
+    def __init__(self, args):
+        self.args = args
+        self.vio_output_counter = 0
+        self.should_quit = False
+        self.index = args.index
+        with open(os.path.join(args.sdk_recording_path, 'calibration.json')) as f:
+            self.calibration_json = json.load(f)
+        num_cams =  len(self.calibration_json['cameras'])
+        if (self.index >= num_cams): raise Exception(f"Too large camera index {self.index}, must be between 0 and {num_cams - 1} (inclusive).")
+        self.replay = spectacularAI.Replay(
+            args.sdk_recording_path,
+            ignoreFolderConfiguration=True,
+            configuration={'useStereo': num_cams != 1, 'useMagnetometer': False, 'parameterSets': ['no-threads']})
+
+        self.imuToCam = np.array(self.calibration_json['cameras'][args.index]['imuToCamera'])
+
+        self.replay.setExtendedOutputCallback(self.on_vio_output)
+        self.replay.setPlaybackSpeed(-1)
+
+    def on_vio_output(self, _, frames):
+        """
+        Callback function that gets called for each VIO output from the replay.
+        """
+        if self.should_quit:
+            self.replay.close()
+            return
+
+        self.vio_output_counter += 1
+
+        # Skip outputs if requested
+        if self.vio_output_counter <= self.args.skip_outputs:
+            if self.vio_output_counter % 100 == 0:
+                 print(f"Skipped {self.vio_output_counter} outputs...")
+            return
+
+        frame = frames[self.index]
+        if frame is None:
+            print("No frame")
+            return
+
+        image = frame.image.toArray()
+
+        # --- Get Pixel from User Click (with zoom) ---
+        zoom_factor = self.args.zoom
+        if zoom_factor < 0.1:
+            print("Warning: Zoom factor is very small, clamping to 0.1.")
+            zoom_factor = 0.1
+
+        if zoom_factor != 1.0:
+            display_image = cv2.resize(image, None, fx=zoom_factor, fy=zoom_factor, interpolation=cv2.INTER_LINEAR)
+        else:
+            display_image = image
+
+        display_image = cv2.cvtColor(display_image, cv2.COLOR_GRAY2BGR)
+
+        window_name = "Double-click target. Press SPACE to confirm selection. Use W, A, S, D keys to fine tune target."
+        cv2.namedWindow(window_name)
+        cv2.setMouseCallback(window_name, mouse_callback)
```

Comment:
Does moving `mouse_callback` inside `on_vio_output` work? Then you could also get rid of the global variables (perhaps replace `global` -> `nonlocal` in the callback)